File: arduinobot_py_examples/arduinobot_py_examples/simple_parameter.py
import rclpy
from rclpy.node import Node
from rcl_interfaces.msg import SetParametersResult
from rclpy.parameter import Parameter

class SimpleParameter(Node):
    def __init__(self):
        super().__init__("simple_parameter")
        self.declare_parameter("simple_int_param", 28)
        self.declare_parameter("simple_string_param", "Matthew")
        
        self.add_on_set_parameters_callback(self.param_change_callback)
        
    def param_change_callback(self, params):
        result = SetParametersResult()
        
        for param in params:
            if param.name == "simple_int_param" and param.type_ == Parameter.Type.INTEGER:
                self.get_logger().info(f"Param simple_int_param changed! New value is {param.value}")
                result.successful = True
            if param.name == "simple_string_param" and param.type_ == Parameter.Type.STRING:
                self.get_logger().info(f"Param simple_string_param changed! New value is {param.value}")
                result.successful = True
        
        return result

def main():
    rclpy.init()
    node = SimpleParameter()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: arduinobot_cpp_examples/src/simple_parameter.cpp
#include "rclcpp/rclcpp.hpp"
#include "rcl_interfaces/msg/set_parameters_result.hpp"

using std::placeholders::_1;

class SimpleParameter : public rclcpp::Node
{
public:
  SimpleParameter() : Node("simple_parameter")
  {
    this->declare_parameter<int>("simple_int_param", 42);
    this->declare_parameter<std::string>("simple_string_param", "Matthew");

    param_callback_handle_ = add_on_set_parameters_callback(std::bind(&SimpleParameter::parameterChangeCallback, this, _1));

  }
private:
    rcl_interfaces::msg::SetParametersResult parameterChangeCallback(const std::vector<rclcpp::Parameter> &parameters)
    {
        rcl_interfaces::msg::SetParametersResult result;
        for (const auto &parameter : parameters)
        {
            if (parameter.get_name() == "simple_int_param" && parameter.get_type() == rclcpp::ParameterType::PARAMETER_INTEGER)
            {
                my_parameter = parameter.as_int();
                result.successful = true;
                RCLCPP_INFO(get_logger(), "Parameter changed. New value is: %d", my_parameter);
            }
            else if (parameter.get_name() == "simple_string_param" && parameter.get_type() == rclcpp::ParameterType::PARAMETER_STRING)
            {
                result.successful = true;
                RCLCPP_INFO(get_logger(), "Parameter changed. New value is: %s", parameter.as_string().c_str());
            }
            else
            {
                result.successful = false;
                RCLCPP_ERROR(get_logger(), "Parameter not found or type mismatch");
            }
        }
        return result;
    }

    int my_parameter;
    OnSetParametersCallbackHandle::SharedPtr param_callback_handle_;
};

int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<SimpleParameter>());
  rclcpp::shutdown();
  return 0;
}